spec.html

```diff
@@ -47057,16 +47246,66 @@ <h1>
+          1. Return ? Call(_reviver_, _holder_, « _name_, _val_, _context_ »).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-static-semantics-shallowestcontainedjsonvalue" type="sdo">
+        <h1>Static Semantics: ShallowestContainedJSONValue ( ): a Parse Node or ~empty~</h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It performs a breadth-first search of the parse tree rooted at the context node, and returns the first node that is an instance of a nonterminal corresponding to a JSON value, or ~empty~ if there is no such node.</dd>
+        </dl>
+        <emu-alg>
+          1. Let _F_ be the active function object.
+          1. Assert: _F_ is a `JSON.parse` built-in function object (see <emu-xref href="#sec-json.parse">JSON.parse</emu-xref>).
+          1. Let _types_ be « |NullLiteral|, |BooleanLiteral|, |NumericLiteral|, |StringLiteral|, |ArrayLiteral|, |ObjectLiteral|, |UnaryExpression| ».
+          1. Let _unaryExpression_ be ~empty~.
+          1. Let _queue_ be « this Parse Node ».
+          1. Repeat, while _queue_ is not empty,
+            1. Let _candidate_ be the first element of _queue_.
+            1. Remove the first element from _queue_.
+            1. Let _queuedChildren_ be *false*.
+            1. For each nonterminal _type_ of _types_, do
+              1. If _candidate_ is an instance of _type_, then
+                1. NOTE: In the JSON grammar, a <code>number</code> token may represent a negative value. In ECMAScript, negation is represented as a unary operation.
+                1. If _type_ is |UnaryExpression|, then
+                  1. Set _unaryExpression_ to _candidate_.
+                1. Else if _type_ is |NumericLiteral|, then
+                  1. Assert: _candidate_ is contained within _unaryExpression_.
+                  1. Return _unaryExpression_.
+                1. Else,
+                  1. Return _candidate_.
+              1. If _queuedChildren_ is *false* and _candidate_ is an instance of a nonterminal and _candidate_ Contains _type_ is *true*, then
+                1. Let _children_ be a List containing each child node of _candidate_, in order.
+                1. Set _queue_ to the list-concatenation of _queue_ and _children_.
+                1. Set _queuedChildren_ to *true*.
+          1. Return ~empty~.
         </emu-alg>
       </emu-clause>
     </emu-clause>
 
+    <emu-clause id="sec-json.rawjson">
+      <h1>JSON.rawJSON ( _text_ )</h1>
+      <p>The `rawJSON` function returns an object representing raw JSON text of a string, number, boolean, or null value.</p>
+      <emu-alg>
+        1. Let _jsonString_ be ? ToString(_text_).
+        1. Throw a *SyntaxError* exception if _jsonString_ is the empty String, or if either the first or last code unit of _jsonString_ is any of 0x0009 (CHARACTER TABULATION), 0x000A (LINE FEED), 0x000D (CARRIAGE RETURN), or 0x0020 (SPACE).
```

Comment:
I noticed that we never use in algorithms the structure "Throw a <type> exception if <condition>", it's usually "If <condition>, throw a <type> exception."

Reply:
Fixed, and added tighter assertions.